src/core/server/http/http_server.test.ts

```diff
@@ -577,6 +577,43 @@ test('exposes route details of incoming request to a route handler', async () =>
     });
 });
 
+describe('conditional compression', () => {
+  test('disables compression when there is a referer', async () => {
+    const { registerRouter, server: innerServer } = await server.setup(config);
+
+    const router = new Router('', logger, enhanceWithContext);
+    router.get({ path: '/', validate: false }, (context, req, res) =>
+      res.ok({ body: 'hello'.repeat(500), headers: { 'Content-Type': 'text/html; charset=UTF-8' } })
```

Comment:
shall we add a comment why we make such a long body?

Reply:
Yup, great idea. I will make it so.